lib/stdlib/src/gb_trees.erl

```diff
@@ -95,6 +95,20 @@
 %%   in tree T, V is the value associated with X in T, and T1 is the
 %%   tree T with key X deleted. Assumes that the tree T is nonempty.
 %%
+%% - lower_entry(K, T): returns {Key, Value} pair, where Key is the
+%%   greatest key strictly less than K, or `none' if no such key exists.
+%%
+%% - higher_entry(K, T): returns {Key, Value} pair, where Key is the
+%%   least key strictly greater than K, or `none' if no such key exists.
+%%
+%% - floor_entry(K, T): returns {Key, Value} pair, where Key is the
+%%   greatest key less than or equal to K, or `none' if no such key
+%%   exists.
+%%
+%% - ceiling_entry(K, T): returns {Key, Value} pair, where Key is the
+%%   least key greater than or equal to K, or `none' if no such key
+%%   exists.
+%%
```

Comment:
1. Why both ceiling (>=) and higher (>)? gb_trees don't have bag semantics so there cannot be two entries with the same key. (Likewise for the other direction.)
2. In Erlang these operations are usually called `next` and `prev`. (C.f. `ets` and `mnesia`. Some other containers unfortunately only have `next`.)
3. `next` already exists, but it takes an iterator not a key (not a serious limitation).

A `prev` operation in `gb_trees` would be a useful addition, IMO.

Reply:
I think both iterator-based API and simple/fast API is a good thing to have. Sometimes we don't want to construct a  log n size iterator just to fetch one entry. It causes more garbage collection.

For the non-iterator API we need to have both ceiling and higher, otherwise it will be hard to use if some user want to achieve both functionality. The naming is because next is used already (also it is unclear if it is inclusive or not)

I'm planning to add ```prev``` function also, just wonder which API would you like?
1. only add prev
2. add prev and iterator_back, iterator_endwith / iterator_floor ...? How should we name them?
3. add prev and a group of new iterator functions, with renaming/deprecating ```iterator_from``` also (I think this function name is really ambiguous)

Thanks!